pkgcheck-action technote (#316)
ropensci/roweb3: content/blog/2022-02-02-pkgcheck-action/index.md

---
slug: pkgcheck-action

> title: 'pkgcheck is now available as a GitHub action'

Reviewer: ```suggestion
title: 'pkgcheck now available as a GitHub action!'
```

date: '2022-02-02'
author:
  - Mark Padgham
  - Jacob Wujciak-Jens
featured: false
description: |
  All packages submitted for peer-review with rOpenSci are checked by our   
  {pkgcheck} package. 
  This post describes a new GitHub action which can be used to run {pkgcheck}.
tags:
  - ropensci-review-bot
  - pkgcheck
  - tech notes
---

## What is pkgcheck?

> The [{pkgcheck} package](https://github.com/ropensci-review-tools/pkgcheck) has

Reviewer: ```suggestion
The [{pkgcheck} package](https://docs.ropensci.org/pkgcheck) has
```

Also, the style guide is to not use curly braces for packages.

Author: I'll remove all of those straight away - thanks

been developed by rOpenSci to automate the process of checking all packages on
submission. The `ropensci-review-bot` automatically runs {pkgcheck} on all submissions, and checks can also be called at any time by editors using the command:

```r
@ropensci-review-bot check package
```

The bot will return a list of checks which should ideally look like this:

> ![example pkgcheck](frictionless-pkgcheck.png)

Reviewer: The alternative text could be a bit more precise. Furthermore, there could be a link to the actual issue comment. Last point, you can use the Hugo figure shortcode (see blog guide https://blogguide.ropensci.org/technical.html#addimage)

Author: I've addressed the first two of those, and done it by linking the entire image to the original issue comment. I have not used handlebars, because I'm not sure whether hugo allows those to be nested in a standard markdown link? @maelle If you know how to achieve what current markdown does in handlebars style, please feel free to commit directly yourself. Thanks!

> To be ready for peer review, {pkgcheck} should return a series of &#10004;, indiciating successful checks, and there shouldn't be any &#10008; symbols, which indicate failed checks. Anybody preparing a package to submit is recommended to ["use pkgcheck"](https://devguide.ropensci.org/authors-guide.html) to confirm that a package is indeed ready to submit. Until now, this has only been possible through locally [installing the package and running the `pkgcheck()` command](https://devguide.ropensci.org/authors-guide.html). Local checks suffer two important disadvantages:

Reviewer: ```suggestion
To be ready for peer review, {pkgcheck} should return a series of &#10004;, indicating successful checks, and there shouldn't be any &#10008; symbols, which indicate failed checks. Anybody preparing a package to submit is recommended to ["use pkgcheck"](https://devguide.ropensci.org/authors-guide.html) to confirm that a package is indeed ready to submit. Until now, this has only been possible through locally [installing the package and running the `pkgcheck()` command](https://devguide.ropensci.org/authors-guide.html). Local checks suffer two important disadvantages:
```

1. Results may only be reproducible on a local system, rather than be generally reproducible; and
2. The checks run by {pkgcheck} include all checks run by [the {rcmdcheck} package](https://r-lib.github.io/rcmdcheck/) which may take quite some time to run.

Just like the [GitHub {rcmdcheck} action](https://github.com/r-lib/actions/blob/v2-branch/examples/check-standard.yaml) which can be easily installed with one of [the `use_github_action_check_...` functions from the `usethis` package](https://usethis.r-lib.org/reference/github_actions.html), {pkgcheck} now also has a GitHub action which overcomes both of these disadvantages through running remotely (on GitHub), and generating reproducible results.

## The GitHub {pkgcheck-action}

> This post is the "official" release announcement of {pkgcheck-action}, a GitHub action which enables checks to be run everytime code is pushed to GitHub. The action was developed by one of us (Jacob), and can be found [on GitHub at ropensci-review-tools/pkgcheck-action](https://github.com/ropensci-review-tools/pkgcheck-action). The easiest way to use this action in your own repository is to install [the {pkgcheck} package](https://docs.ropensci.org/pkgcheck/#installation), and run [the `use_github_action_pkgcheck()` function](https://docs.ropensci.org/pkgcheck/reference/use_github_action_pkgcheck.html). That will place a new file in the `.github/workflows` directory of your package called "pkgcheck.yaml", an example of which is shown on [the main REAMDE of the {pkgcheck-action} repository](https://github.com/ropensci-review-tools/pkgcheck-action#usage). The action consists of the single job expressed in the final lines:

Reviewer: ```suggestion
This post is the "official" release announcement of {pkgcheck-action}, a GitHub action which enables checks to be run everytime code is pushed to GitHub. The action was developed by one of us (Jacob), and can be found [on GitHub at ropensci-review-tools/pkgcheck-action](https://github.com/ropensci-review-tools/pkgcheck-action). The easiest way to use this action in your own repository is to install [the {pkgcheck} package](https://docs.ropensci.org/pkgcheck/#installation), and run [the `use_github_action_pkgcheck()` function](https://docs.ropensci.org/pkgcheck/reference/use_github_action_pkgcheck.html). That will place a new file in the `.github/workflows` directory of your package called "pkgcheck.yaml", an example of which is shown on [the main README of the {pkgcheck-action} repository](https://github.com/ropensci-review-tools/pkgcheck-action#usage). The action consists of the single job expressed in the final lines:
```

```yaml
jobs:
  check:
    runs-on: ubuntu-latest
    steps:

>       - uses: ropensci-review-tools/pkgcheck-action@main

Reviewer: no v1? why not have a release? wouldn't it be more standard?

Author: @assignUser explains in the README. It's a docker-based action, and the docker image is updated via CRON, including all latest updates of all R deps. That tag ensures that the latest docker container is always used. Putting a fixed tag will fix the docker image, and so the whole action will quickly become outdated.

Reviewer 2: The most important bit being that the docker image will be updated every time {pkgcheck} is updated, so people will always pass the most up-to-date checks etc.. and not get "false positives" and prematurely submit to review.

If we want to publish the action the Github Marketplace we need to do a release regardless but that should not be an issue.

```

> There are several parameters also described in that main README, which can either be passed as parameters to the `use_github_action_pkgcheck()` function, or added to the YAML workflow file as also described in the README. Once the action has been created in your repository, it can be activated by adding it in a git commit and pushing to GitHub.

Reviewer: link to pkgdown docs of that function?

Author: done, thanks

### Posting {pkgcheck} results to a GitHub issue

> In addition to the workflow printing the full details returned by the main {pkgcheck} function, the output of {pkgcheck} can also be posted to a GitHub issue, where it will appear precisely like the example shown above. Once the workflow has finished, you'll receive a notification, and be able to view the check results (with a `summary-only` parameter determining whether only a summary of results like shown above will be posted to the issue, or whether full results will be posted which include the summary followed by quite a bit more detail). The workflow will only succeed when all {pkgcheck} checks have been passed.

Reviewer: Example issue?

Reviewer: Preferably for a toy package repo

Author: @assignUser can you please put a link to a example issue in a demo repo?

Reviewer 2: We could create an example/testing repo under `ropensci-review-tools` so branding is consistent. 
Otherwise I could over this: https://github.com/assignUser/octolog/issues/4
Let me know :)
